src/app/directives/app-role-required.directive.ts

```diff
@@ -10,8 +11,8 @@ export class RoleRequiredDirective implements OnInit {
   userRolesRequired: string[] | undefined;
 
   @Input()
-  set appRoleRequired(roles: Roles) {
-    this.userRolesRequired = roles?.split(',');
+  set appRoleRequired(roles: Roles | Roles[]) {
+    Array.isArray(roles) ? (this.userRolesRequired = uniq(roles.flatMap(role => role.split(',')))) : (this.userRolesRequired = roles?.split(','));
```

Comment:
I don't think you need this split anymore. It was there because multiple roles used to be declared like 'role1, role2' or that was the plan

Reply:
Is still is like that 😄

Reply:
Then why add this other functionality?

Reply:
Because ive extended it aswell because now some journeys can starts from having one of x roles in an array

Reply:
I did want to use a polymorphic approach but the root Declaration can only take one identifier

Reply:
https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle#:~:text=In%20object%2Doriented%20programming%2C%20the,without%20modifying%20its%20source%20code.